account_invoice_show_currency_rate/models/account_move.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    currency_rate_amount = fields.Float(
+        string="Rate amount",
+        compute="_compute_currency_rate_amount",
+        store=True,
+        digits=0,
+    )
+
+    @api.depends("date", "state", "company_id", "currency_id")
+    def _compute_currency_rate_amount(self):
+        """ It's necessary to define value according to some cases:
+        - Case A: Currency is equal to company currency (Value = 1)
+        - Case B: Move exist previously (posted) and get real rate according to lines
+        - Case C: Get expected rate (according to date) to show some value in creation.
+        """
+        for item in self:
+            if (
+                item.state == "posted"
+                and item.currency_id != item.company_id.currency_id
+            ):
+                amount_currency_positive = sum(
+                    item.line_ids.filtered(lambda x: x.amount_currency > 0).mapped(
+                        "amount_currency"
+                    )
+                )
+                total_debit = sum(item.line_ids.mapped("debit"))
+                item.currency_rate_amount = amount_currency_positive / total_debit
+            elif item.date and item.currency_id != item.company_id.currency_id:
+                rates = item.currency_id._get_rates(item.company_id, item.date)
+                item.currency_rate_amount = rates.get(item.currency_id.id)
+            else:
+                item.currency_rate_amount = 1
```

Comment:
Better to set 0 than 1, and at the beginning of the compute method for all in one shot: `self.currency_rate_amount = 0`

Reply:
I'm now getting the reason for that. Although it's going to be hidden, `1` is OK, yes. Do the other thing of putting it at the beginning in one shot.